File: src/EditorFeatures/CSharpTest/Outlining/AbstractOutlinerTests.cs
// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.

using System;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.Editor.Implementation.Outlining;
using Xunit;

namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Outlining
{
    public abstract class AbstractOutlinerTests
    {
        internal void AssertRegion(OutliningSpan expected, OutliningSpan actual)
        {
            Assert.Equal(expected.TextSpan.Start, actual.TextSpan.Start);
            Assert.Equal(expected.TextSpan.End, actual.TextSpan.End);
            Assert.Equal(expected.HintSpan.Start, actual.HintSpan.Start);
            Assert.Equal(expected.HintSpan.End, actual.HintSpan.End);
            Assert.Equal(expected.BannerText, actual.BannerText);
            Assert.Equal(expected.AutoCollapse, actual.AutoCollapse);
        }

        protected SyntaxTree ParseCode(string code)
        {
            return SyntaxFactory.ParseSyntaxTree(code);
        }

        protected string StringFromLines(params string[] lines)
        {
            return string.Join(Environment.NewLine, lines);
        }

        protected SyntaxTree ParseLines(params string[] lines)
        {
            var code = StringFromLines(lines);
            return ParseCode(code);
        }
    }
}

File: src/EditorFeatures/VisualBasicTest/Outlining/AbstractOutlinerTests.vb
' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.

Imports Microsoft.CodeAnalysis.Editor.Implementation.Outlining

Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.Outlining
    Public MustInherit Class AbstractOutlinerTests
        Friend Sub AssertRegion(expected As OutliningSpan, actual As OutliningSpan)
            Assert.Equal(expected.TextSpan.Start, actual.TextSpan.Start)
            Assert.Equal(expected.TextSpan.End, actual.TextSpan.End)
            Assert.Equal(expected.HintSpan.Start, actual.HintSpan.Start)
            Assert.Equal(expected.HintSpan.End, actual.HintSpan.End)
            Assert.Equal(expected.BannerText, actual.BannerText)
            Assert.Equal(expected.AutoCollapse, actual.AutoCollapse)
            Assert.Equal(expected.IsDefaultCollapsed, actual.IsDefaultCollapsed)
        End Sub
    End Class
End Namespace
